### DISCUSSION TITLE
Providing `\Closure` type for variable

### DISCUSSION THREAD

**Discussion by Rizky92** [vimeo/psalm#10099] [Q&A]
I have structure of a class that looks like this
https://psalm.dev/r/6b1614ce42

I wanted to provide typings for closure stored in `$data`, so that later on I could call `$data(new B)->a()`. How do I provide typings for it? I tried template but I think I don't quite understand how it works.

**Comment by andrew-demb** [comment]
https://psalm.dev/r/16bbd9d181

Here's an expanded example for your case.
